torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1454,6 +1454,252 @@ class AutogradLazyBackwardCompileInfo:
     saved_compile_context: Optional[CompileContext]
 
 
+def _raise_if_functorch_active():
+    # not ideal but prevent the user from seeing a nasty traceback - See #138422
+    stack = torch._C._functorch.peek_interpreter_stack()
+    torch._check(
+        stack is None,
+        lambda: (
+            "It looks like you're trying to call a compiled backward function within vmap/grad/vjp, "
+            "which isn't supported. Try wrapping vmap inside torch.compile, or skip compiling the "
+            "backward function."
+        ),
+    )
+
+
+# NOTE: this function must be torch._dynamo.allow_in_graph-able. Non tensor/symnode inputs must be constants.
+def _backward_prologue_functional(
```

Comment:
bigdiff to help review: https://www.internalfb.com/intern/bigdiff/?original=P1694369532&modified=P1694369957&language=json

Reply:
I think I was expecting a `@torch._dynamo.allow_in_graph` above this function - is that not part of this PR?

Reply:
also thank you for the bigdiff

Reply:
it'll be in the FCA PR, also we'll never directly torch._dynamo.allow_in_graph this function because that would lift all the function inputs as graph inputs. We allow_in_graph a wrapper and smuggle in the const fold arguments metadata and maybe_subclass_metadata